media-api/app/lib/elasticsearch/QueryBuilder.scala

```diff
@@ -11,8 +11,21 @@ import com.sksamuel.elastic4s.requests.common.Operator
 import com.sksamuel.elastic4s.requests.searches.queries.Query
 import com.sksamuel.elastic4s.requests.searches.queries.matches.{MultiMatchQuery, MultiMatchQueryBuilderType}
 import lib.querysyntax._
-class QueryBuilder(matchFields: Seq[String], overQuotaAgencies: () => List[Agency]) extends ImageFields with GridLogging {
+import lib.MediaApiConfig
 
+class QueryBuilder(matchFields: Seq[String], overQuotaAgencies: () => List[Agency], config: MediaApiConfig) extends ImageFields with GridLogging {
+
+  def getAdvancedFieldPath(field: String): String = {
+
+    val fileMetadataPath = config.fieldAliasConfigs.find(_.alias == field) match {
+      case Some(x) => Some(x.elasticsearchPath)
+      case None => None
+    }
+    val simplePath = getFieldPath(field)
+
+    if(fileMetadataPath != None) fileMetadataPath.get
+    else simplePath
+  }
```

Comment:
This function is needlessly complicated. I'd also suggest renaming it to something like `resolveFieldPath`.
```suggestion
  def getAdvancedFieldPath(field: String): String = {
    val fileMetadataPath = config.fieldAliasConfigs.find(_.alias == field) match {
      case Some(x) => x.elasticsearchPath
      case None => getFieldPath(field)
    }
  }
```

Reply:
I agree with you. I also looked into doing that in [ImageFields](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/lib/ImageFields.scala#L50-L58) but had to revert back to this as it just seems more straight forward and does not require changing multiple files. Will simplify the function now.